### DISCUSSION TITLE
How to change api_tokens id from integer to string and fill it with UUID

### DISCUSSION THREAD

**Discussion by frama21** [adonisjs/core#3003] [Help]
can i change api_tokens table id from integer to string? and then generating UUID when user login or register?

i change the api_tokens table id from integer to string, then i try to create user and i got the error like this 

```bash 
 "insert into `api_tokens` (`created_at`, `expires_at`, `name`, `token`, `type`, `user_id`) values ('2021-08-06 12:09:35', '2021-08-16 12:09:35', 'Opaque Access Token', '5d13f47b11402c804e65f06f3572ff7eecbc3e8a3a879af68b9385a5722df0c0', 'opaque_token', '74d7d4ed-47fa-4673-bc8d-9005e9ddd426') - ER_NO_DEFAULT_FOR_FIELD: Field 'id' doesn't have a default value"
 ```
 
this my api_tokens migration up
 ```js
 public async up() {
    this.schema.createTable(this.tableName, (table) => {
      table.string('id').primary()
      table.string('user_id', 255)
      table.string('name').notNullable()
      table.string('type').notNullable()
      table.string('token', 64).notNullable()

      /**
       * "useTz: true" utilizes timezone option in PostgreSQL and MSSQL
       */
      table.timestamp('expires_at', { useTz: true }).nullable()
      table.timestamp('created_at', { useTz: true }).notNullable()
    })
  }
 ```

i use adonis 5 and MySQL as database
if i wrong to implement it please tell me, or if you can give me some example code it's really apreciate it thank you so much

**Comment by feniljariwala82** [comment]
Search this in the discussion forum, you will get your answer. 
**[Adonis 5] Using UUID in the tables**

**Reply by frama21** [reply]
in my case when user was autheticated, the data of token or user login was store in table api_tokens.
table api_tokens is generate automatic in first time when i invoke the package of authentication from adonisjs
the generate of table api_tokens migration is fill like this by default then i change some values to :

```js
public async up() {
   this.schema.createTable(this.tableName, (table) => {
     table.string('id').primary() // i want change this to string from incremental because i want make this set to store UUID
     table.string('user_id', 255) // this set to string because my user id is UUID
     table.string('name').notNullable()
     table.string('type').notNullable()
     table.string('token', 64).notNullable()

     /**
      * "useTz: true" utilizes timezone option in PostgreSQL and MSSQL
      */
     table.timestamp('expires_at', { useTz: true }).nullable()
     table.timestamp('created_at', { useTz: true }).notNullable()
   })
 }
 ```
 
 when i change the ```id``` column in ```api_tokens``` migration to string, other data was storing success without error but the ```id``` is null 

FYI: api_tokens not have models after invoke the package or maybe i must create the api_tokens manual?
